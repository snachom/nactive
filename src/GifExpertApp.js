import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

import './gif-expert.css'

export const GifExpertApp = () => {

  const [categories, setCategories] = useState([ 'Charlie Brown' ]);

  return ( 
    <>
      <div className="title-wrapper">
        <h2 className="app-title"> Giff App </h2>
        <hr />
      </div>
      <AddCategory setCategories={ setCategories } />

      <ol>
        { 
          categories.map( category => 
            <GifGrid 
              key={ category }
              category={ category } 
            />
          )
        }
      </ol>
    </>
  )
};
