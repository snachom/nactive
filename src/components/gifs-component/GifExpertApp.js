import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

import './gif-expert.css'

// export const GifExpertApp = ({ defaultCategories = ['Charlie Brown'] }) => {
export const GifExpertApp = ({ defaultCategories = ['Charlie Brown'] }) => {

  // const [categories, setCategories] = useState([ 'Charlie Brown' ]);
  const [categories, setCategories] = useState(defaultCategories);

  return ( 
    <>
      <div className="title-wrapper">
        <h2 className="app-title"> Giff App </h2>
        <hr />
      </div>
      <AddCategory setCategories={ setCategories } />

      <ol>
        { 
          categories.map( category => 
            <GifGrid 
              key={ category }
              category={ category } 
            />
          )
        }
      </ol>
    </>
  )
};
