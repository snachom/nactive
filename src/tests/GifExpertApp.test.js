import React from 'react'
import {shallow} from 'enzyme'
import '@testing-library/jest-dom'
import {GifExpertApp} from '../GifExpertApp'

describe('Pruebas sobre GifExpertApp', () => {

  const wrapper = shallow(<GifExpertApp />)

  test('GifExpertApp', () => {

    expect(wrapper).toMatchSnapshot()

  })

  test('debe mostrar una lista de categorÃ­as', () => {

    const categories = ['Snoopy', 'Charlie Brown']
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toBe(categories.length)
    
  })

})