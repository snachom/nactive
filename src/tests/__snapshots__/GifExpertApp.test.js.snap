// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre GifExpertApp GifExpertApp 1`] = `
<Fragment>
  <div
    className="title-wrapper"
  >
    <h2
      className="app-title"
    >
       Giff App 
    </h2>
    <hr />
  </div>
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Pruebas sobre GifExpertApp debe mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <div
    className="title-wrapper"
  >
    <h2
      className="app-title"
    >
       Giff App 
    </h2>
    <hr />
  </div>
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="Snoopy"
      key="Snoopy"
    />
    <GifGrid
      category="Charlie Brown"
      key="Charlie Brown"
    />
  </ol>
</Fragment>
`;
